src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const fromParts = fromFormat.slice(0, -1);
+  const toParts = toFormat.slice(0, -1);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateObj = {};
+
+  fromParts.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  const formattedDate = toParts
+    .map((part) => {
+      if (part === 'YYYY' && dateObj['YY']) {
```

Comment:
Be cautious with direct property access on 'dateObj' using bracket notation with a string literal. If 'dateObj' does not have a property 'YY', this will return 'undefined', which could lead to unexpected behavior. It would be safer to check if the property exists before attempting to use it.